app/controllers/admin/application_controller.rb

```diff
@@ -1,12 +1,12 @@
 class Admin::ApplicationController < ApplicationController
-  before_action :authorized?
+  before_action :authenticate_user!
+  before_action :authenticate_admin!
 
   private
 
-  def authorized?
-    return if current_user.admin?
+  def authenticate_admin!
+    return if current_user.try(:admin?)
 
-    flash[:error] = t('admin.forbidden_error')
-    redirect_to(root_path)
+    redirect_to '/', alert: 'Not authorized.'
```

Comment:
better to use `root_path` instead of `'/'`

Reply:
and move the text to locale

Reply:
moved.